### DISCUSSION TITLE
v0.10.0-preview1

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#439] [Announcements]
v0.10 的第一个预览版现已释出，使用中遇到问题请提交 issue。注意 v0.10 仍在开发中，以下功能尚未完成：

* 本地化（目前只支持中文）
* 仍在征集新图标 #370 
* 自动更新

v0.10 最低操作系统版本为 Win10 v1903。无需 .NET 运行时。

**新功能**

![image](https://user-images.githubusercontent.com/34770031/196185400-bd55b110-e846-4702-9479-27499d880f9d.png)

1. 全新的基于 WinUI 的用户界面，支持明暗主题和 Mica 背景
    * Mica 背景仅限 Win11 22H2+
2. 特定窗口（游戏）的配置 #423 #385 #130 #61
    * 可直接在配置界面启动游戏
3. 支持在用户界面中配置缩放模式
4. 支持“默认降采样效果”，窗口缩放后如果大于屏幕尺寸将自动应用此效果
    * 默认为 Bicubic
5. 支持导出/导入缩放模式，可用于分享配置
    * 支持导入旧版缩放配置（ScaleModels.json）
6. 新的修改快捷键的用户界面，和 PowerToys 体验相同
7. 便携模式，开启后会将配置文件保存在本地文件夹
8. 升级了常驻后台功能，开启后始终在系统托盘中显示图标，关闭主窗口不会退出程序
9. 开机启动时不会弹出 UAC #327 
10. 支持自定义倒计时时长 #258 
11. 系统截图时自动隐藏光标 #394 
    * 卡顿时可能不可靠
12. “保留标题栏”选项
    * 仅限 Graphics Capture 和 Desktop Duplication 捕获模式
13. 支持自定义光标缩放
14. “内联效果参数”选项。开启后将稍微提高性能，但每次修改参数将需重新编译该效果

**效果变更**
1. 现已支持文件夹分组
2. CRT_Geom 默认不再模拟隔行扫描 #415
3. 部分基于机器学习的效果现在有了更平滑的边缘 #398

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.10.0-preview1'>v0.10.0-preview1</a>.</em>

**Comment by greezmann11** [comment]
It causes my games to lag anytime I activate it

**Reply by Blinue** [reply]
There are many reasons for performance degradation. If you think it's a bug, please file an issue.

**Reply by greezmann11** [reply]
How do I do that

**Reply by palxex** [reply]
@Blinue 同感，刚测试了一下感觉输入延迟明显增大，体现在游戏上是手感变得笨重了，而fps变化不大。但不太清楚输入延迟应该如何量化？尽量保持了配置与0.9时一样，垂直同步开，额外延迟关，禁用DirectFlip关，测试滤镜是从Integer Scale 2x改成的Integer Scale 1x（只将所有2改成1）。
另外提个问题，0.10preview1修复了rtss在scaled texture上的渲染，但在因为比例问题出现的黑屏上每帧似乎没有清空（游戏本来的内容上没问题），这导致在黑屏上其内容逐渐堆叠至无法看清。
![image](https://user-images.githubusercontent.com/58222/198876142-0f957704-4dd3-434d-b976-d915195c08d0.png)
因为最近的版本我无法编译，无法确认问题在当前dev HEAD上是否仍然存在。

**Reply by hooke007** [reply]
在action里获取

**Reply by palxex** [reply]
> 在action里获取

谢谢。
经测试提交b348d3e的artifacts，渲染堆积问题无变化，仍如上图。
此外该提交的跟手情况相比preview1改善明显，跟0.9.1已感觉不出区别。

**Reply by Blinue** [reply]
> 另外提个问题，0.10preview1修复了rtss在scaled texture上的渲染，但在因为比例问题出现的黑屏上每帧似乎没有清空（游戏本来的内容上没问题），这导致在黑屏上其内容逐渐堆叠至无法看清。

为了提高性能，渲染之前不会清空前一帧的内容。因为每次只渲染同一个区域所以不会有问题。或许应该增加一个选项来兼容 RTSS。

**Reply by crestcere** [reply]
> It causes my games to lag anytime I activate it

I got same issue too with dead by daylight. Manage to outcome with disabling this setting.

![Snipaste_2022-11-04_23-01-44](https://user-images.githubusercontent.com/31207969/200064524-0413c3ab-43d9-41af-8654-d2a0f05c4464.png)

**Reply by greezmann11** [reply]
> > It causes my games to lag anytime I activate it
> 
> I got same issue too with dead by daylight. Manage to outcome with disabling this setting.
> 
> ![Snipaste_2022-11-04_23-01-44](https://user-images.githubusercontent.com/31207969/200064524-0413c3ab-43d9-41af-8654-d2a0f05c4464.png)

**Reply by greezmann11** [reply]
Which settings did u disable. Put it in writing so I can translate it.

**Reply by palxex** [reply]
> Which settings did u disable. Put it in writing so I can translate it.

"Adjust Cursor Speed while Zoomed" in English.

**Reply by greezmann11** [reply]
Thanks

**Reply by Blinue** [reply]
@palxex 已经修复和 RTSS 的兼容性

**Comment by hooke007** [comment]
一按导入/导入旧版直接闪退。log没有记录相关信息
```
2022-10-18 21:30:28.321|info|XamlApp.cpp:Initialize|程序启动
	版本：0.10.0-preview1
2022-10-18 21:30:29.799|info|XamlApp.cpp:Run|程序退出
2022-10-18 21:30:29.816|info|XamlApp.cpp:Initialize|程序启动
	版本：0.10.0-preview1
2022-10-18 21:30:30.049|info|MainPage.cpp:_UpdateTheme|当前主题：浅色
```

**Reply by hooke007** [reply]
测试了半天终于找到复现方法了。
清理配置文件，重启，点击导入——正常
设置为管理员模式，重启，点击导入——闪退

**Reply by Blinue** [reply]
谢谢。我找到问题所在了 https://github.com/microsoft/WindowsAppSDK/issues/2504

**Reply by Blinue** [reply]
已修复

**Comment by 0x4E69676874466F78** [comment]
@Blinue please ping when localization support is available.

**Reply by Blinue** [reply]
Localization will not be supported until all other features are implemented. Maybe weeks.

**Comment by hooke007** [comment]
几个界面设计上的看法

改成允许批量处理的形式，是不是比每个选项后的二级菜单内的删除，要更加方便点
<img src="https://user-images.githubusercontent.com/41094733/196788081-70cf9b06-f1eb-49fd-8c08-a98965dcfe41.jpg" width=400>

添加效果，先是二级菜单靠前，然后才是按字母排列。。。应该是bug

<img src="https://user-images.githubusercontent.com/41094733/196789255-56cf80c0-2050-4c5e-95ee-1ae78dcb1cb3.jpg" height=400>

**Reply by Blinue** [reply]
> 改成允许批量处理的形式，是不是比每个选项后的二级菜单内的删除，要更加方便点

我曾经也考虑过这个方案，但这会让用户界面变得复杂，另一个原因是我以为删除功能是不常用到的。可以把删除的层级提高，但我不喜欢界面上有太多按钮
![image](https://user-images.githubusercontent.com/34770031/196861077-072427cf-97da-4120-b3a0-c48d06b7394c.png)

> 添加效果，先是二级菜单靠前，然后才是按字母排列。。。应该是bug

这是刻意设计的，为了模拟文件夹。原本是这样的
![image](https://user-images.githubusercontent.com/34770031/196861949-110c838a-52d7-4ec4-a1eb-96473834a9c1.png)

**Reply by hooke007** [reply]
模拟文件夹，，额看不懂，我作为用户的想法是根据首字母快速检索到对应位置（而且维基里的效果介绍顺序也是如此
把具有变体的效果前置，这需要用户提前熟悉每个效果才能对应位置

**Reply by Blinue** [reply]
现在效果是分组的，所以在查找一个效果前首先要找到它所在的组。有些效果不在任何组里就放在最后。这像是在资源管理器里查找一个文件，要首先打开它所在的文件夹。

**Reply by hooke007** [reply]
如果我理解正确的话就是在模拟打开effect文件夹浏览？恩。。我个人角度看很奇怪，很少有人会去研究程序的文件树结构

**Reply by Blinue** [reply]
组就是文件夹的概念，支持嵌套。现在的排序方式和资源管理器保持一致，是我的个人习惯。如果有更多人反馈，我会把它改掉。

**Comment by neavo** [comment]
简单试用了一下，UI上有两个小建议：
1、缩放模式建议支持鼠标拖拽排序，选项比较多时，一下一下的去点击 “上移” “下移” 其实也挺麻烦的
2、添加效果时候选列表中每一组的组内排序最好可以按照对性能消耗的大小来排序，比如S在前面，L在后面
3、在添加效果的界面，最好可以加入对每个效果的说明，或者加入一个链接点击跳转到wiki页以方便了解各效果的信息

**Reply by Blinue** [reply]
> 缩放模式建议支持鼠标拖拽排序，选项比较多时，一下一下的去点击 “上移” “下移” 其实也挺麻烦的

我很乐意增加操作的便捷性，但和批量处理一样，这会大大增加 UI 的复杂度。每个缩放模式都是可以展开的，因此可以接受鼠标点击，这种情况下 WinUI 无法处理拖拽。移动缩放模式并不是常用的操作，为了它增加大量代码是不值得的。

未来可能会增加一个批处理模式，有一个特殊的 UI，可以批量删除和拖拽，至少在 v0.10 不会实现。

> 添加效果时候选列表中每一组的组内排序最好可以按照对性能消耗的大小来排序，比如S在前面，L在后面

我也想要这个功能。目前是按照字母排序，实现这个功能需要增加特殊的排序机制。

> 在添加效果的界面，最好可以加入对每个效果的说明，或者加入一个链接点击跳转到wiki页以方便了解各效果的信息

添加效果目前只是一个菜单，未来可能会把它变成一个弹窗，这是一个重大功能，v0.10 不会实现。

**Reply by neavo** [reply]
另外报个BUG，设置开机自动启动后，开机启动时会显示主界面而不是最小化模式启动

**Reply by Blinue** [reply]
打开“开机启动时最小化到系统托盘”

**Reply by neavo** [reply]
> 打开“开机启动时最小化到系统托盘”

找到了，原来隐藏在折叠菜单里 （手动捂脸

**Reply by neavo** [reply]
继续提UI改进的建议：建议在任务栏菜单中加入手动切换配置的快捷操作选项，跟自动匹配窗口功能搭配使用应该就可以覆盖绝大部分需要切换配置的使用场景了

![未命名文件](https://user-images.githubusercontent.com/998676/197680216-886ca226-f8cc-4e8c-b771-93625433bdec.jpg)

**Reply by Blinue** [reply]
好主意，请提一个 issue 用于追踪

**Reply by Blinue** [reply]
> 添加效果时候选列表中每一组的组内排序最好可以按照对性能消耗的大小来排序，比如S在前面，L在后面

已实现

**Comment by regs01** [comment]
Can't get it working with  Star Citizen. With or without app profile. Doesn't react to start and hotkey. I tried changing hotkey to Ctrl+Alt+F11, considering CIG forbad using Windows key with Star Citizen, but still no reaction. Tried also as an admin. No luck also. 0.9 is working.

**Reply by Blinue** [reply]
Have you tried "Scale after 5s"？Please attach the logs, they are helpful for debugging.

**Reply by regs01** [reply]
Oh, there is button for that. Thought it's some kind of text field. It's working with it.
Profile, though, doesn't work. There is nothing in logs about that.

```
2022-10-28 22:47:19.793|info|XamlApp.cpp:Initialize|程序启动
	版本：0.10.0-preview1
2022-10-28 22:47:20.057|info|MainPage.cpp:_UpdateTheme|当前主题：浅色
2022-10-28 22:50:12.651|info|MagApp.cpp:_RegisterWndClasses|已注册主窗口类
2022-10-28 22:50:12.651|info|MagApp.cpp:_RegisterWndClasses|已注册 DDF 窗口类
2022-10-28 22:50:12.654|info|MagApp.cpp:_CreateHostWnd|主窗口尺寸：3840x2160
2022-10-28 22:50:12.655|info|DeviceResources.cpp:Initialize|可变刷新率支持：是
2022-10-28 22:50:12.661|info|DeviceResources.cpp:LogAdapter|当前图形适配器：
	VendorId：0x10de
	DeviceId：0x2482
	描述：NVIDIA GeForce RTX 3070 Ti
2022-10-28 22:50:12.689|info|DeviceResources.cpp:Initialize|已创建 D3D Device
	功能级别：11.1
2022-10-28 22:50:12.696|info|DeviceResources.cpp:_CreateSwapChain|Hardware Composition 支持：是
2022-10-28 22:50:12.696|info|DeviceResources.cpp:_CreateSwapChain|Multiplane Overlay 支持：是
2022-10-28 22:50:12.697|info|MagApp.cpp:_InitFrameSource|当前捕获模式：Graphics Capture
2022-10-28 22:50:12.822|info|GraphicsCaptureFrameSource.cpp:Initialize|GraphicsCaptureFrameSource 初始化完成
2022-10-28 22:50:12.822|info|MagApp.cpp:_InitFrameSource|源窗口尺寸：1920x1080
2022-10-28 22:50:12.823|info|Win32Utils.cpp:ReadFile|读取文件：cache\NIS#NIS_13791923dcf1e9e2b
2022-10-28 22:50:12.830|info|EffectCacheManager.cpp:Load|已读取缓存 cache\NIS#NIS_13791923dcf1e9e2b
2022-10-28 22:50:12.831|info|Renderer.cpp:CompileEffect|编译 NIS\NIS.hlsl 用时 7.3630004 毫秒
2022-10-28 22:50:12.840|info|Win32Utils.cpp:ReadFile|读取文件：assets\NotoSansSC-Regular.otf
2022-10-28 22:50:12.870|info|CursorManager.cpp:Initialize|CursorManager 初始化完成
2022-10-28 22:50:12.875|info|CursorManager.cpp:GetCursorTexture|已解析光标：0x247609ff
	类型：Color
2022-10-28 22:50:40.834|info|CursorManager.cpp:GetCursorTexture|已解析光标：0x10003
	类型：Color
2022-10-28 22:50:40.863|info|Renderer.cpp:_CheckSrcState|前台窗口已改变
2022-10-28 22:50:40.863|info|Renderer.cpp:Render|源窗口状态改变，退出全屏
2022-10-28 22:50:41.932|info|MagApp.cpp:_CreateHostWnd|主窗口尺寸：3840x2160
2022-10-28 22:50:41.932|info|DeviceResources.cpp:Initialize|可变刷新率支持：是
2022-10-28 22:50:41.939|info|DeviceResources.cpp:LogAdapter|当前图形适配器：
	VendorId：0x10de
	DeviceId：0x2482
	描述：NVIDIA GeForce RTX 3070 Ti
2022-10-28 22:50:41.974|info|DeviceResources.cpp:Initialize|已创建 D3D Device
	功能级别：11.1
2022-10-28 22:50:41.976|info|DeviceResources.cpp:_CreateSwapChain|Hardware Composition 支持：是
2022-10-28 22:50:41.976|info|DeviceResources.cpp:_CreateSwapChain|Multiplane Overlay 支持：是
2022-10-28 22:50:41.976|info|MagApp.cpp:_InitFrameSource|当前捕获模式：Graphics Capture
2022-10-28 22:50:42.058|info|GraphicsCaptureFrameSource.cpp:Initialize|GraphicsCaptureFrameSource 初始化完成
2022-10-28 22:50:42.058|info|MagApp.cpp:_InitFrameSource|源窗口尺寸：1920x1080
2022-10-28 22:50:42.058|info|EffectCacheManager.cpp:_LoadFromMemCache|已读取缓存 cache\NIS#NIS_13791923dcf1e9e2b
2022-10-28 22:50:42.058|info|Renderer.cpp:CompileEffect|编译 NIS\NIS.hlsl 用时 0.19700001 毫秒
2022-10-28 22:50:42.064|info|CursorManager.cpp:Initialize|CursorManager 初始化完成
2022-10-28 22:50:42.069|info|CursorManager.cpp:GetCursorTexture|已解析光标：0x247609ff
	类型：Color
2022-10-28 22:50:42.662|info|CursorManager.cpp:GetCursorTexture|已解析光标：0x10003
	类型：Color
2022-10-28 22:50:42.681|info|Renderer.cpp:_CheckSrcState|前台窗口已改变
2022-10-28 22:50:42.681|info|Renderer.cpp:Render|源窗口状态改变，退出全屏

```

**Comment by yakymoto** [comment]
Hey, @Blinue I must admit - latest release v0.10 is awesome. I've spend some time to translate each part to English. I hope it is going to be useful. Keep up the good work!

You can copy text from my Photoshop files, please follow the link - mega.nz/folder/DVowWRwS#tiO--wINSzDnXAc7mHgv-A

yakymoto.com - contact me if you have any questions!

![1_1_Home](https://user-images.githubusercontent.com/2016611/200143680-6d795919-0336-4601-88e0-b771ef3b49f1.png)
![1_2_Activation-shortcut-popup](https://user-images.githubusercontent.com/2016611/200143673-854528eb-ca70-4a0e-993b-4910493a4ba3.png)
![2_Scaling-configuration](https://user-images.githubusercontent.com/2016611/200143675-0afba6f6-fbd6-4fe8-9fc4-f08c08615fad.png)
![3_App-defaults](https://user-images.githubusercontent.com/2016611/200143676-abec1aac-77fc-4481-9c44-4d7459838924.png)
![4_Add-new-config](https://user-images.githubusercontent.com/2016611/200143677-270b5bb3-1d8b-4807-b384-5c94fbd59e06.png)
![5_About](https://user-images.githubusercontent.com/2016611/200143678-d13c4710-af2b-4532-aeca-d6b4e3c6a889.png)
![6_Settings](https://user-images.githubusercontent.com/2016611/200143679-9df5cac0-8ff6-4f79-ac21-d94236a9b6d0.png)

**Reply by Blinue** [reply]
Thanks a lot! I'll be working on localization soon.
 I'm thinking about how to enable people with no programming background to contribute translations.

**Reply by hooke007** [reply]
我之前参与过的一个项目使用 https://hosted.weblate.org/ 来允许任何人都能加入多国语言的翻译
<img src="https://user-images.githubusercontent.com/41094733/200156735-b64df5c9-64bf-4424-b88d-5dec040f5a75.jpg" width=500>

或者mpc这些经典多国语言的项目使用的 Transifex

**Reply by Blinue** [reply]
谢谢，我会研究一下

**Reply by anivd84** [reply]
Hello, @Blinue, will it be possible to automatically start scaling after starting the program/game after n-sec? it seems that there is no such thing now, but, as it seems to me, it would not hurt many....

**Reply by Blinue** [reply]
> Hello, @Blinue, will it be possible to automatically start scaling after starting the program/game after n-sec? it seems that there is no such thing now, but, as it seems to me, it would not hurt many....

Tracked in #193

**Comment by FavoriteKyoani** [comment]
在macOS用Parallels Desktop虛擬機使用這款軟件時截圖還是會把被放大了的鼠標截進去，但如果關閉「繪製鼠標｜功能的話鼠標就直接不顯示了，鼠標操作手感也很怪，很不跟手，以及全屏後也只是周圍變黑並且把原來的畫面置中了而已，0.9版本是沒有這個問題的，希望能改善

**Reply by Blinue** [reply]
1. 不支持虚拟机
2. 黑边问题见 https://github.com/Blinue/Magpie/issues/448#issuecomment-1308045339

**Reply by FavoriteKyoani** [reply]
好的，謝謝🙏

**Reply by FavoriteKyoani** [reply]
我在虛擬機上可以正常運行，但就是鼠標在截圖的時候會被截進去，使用的截圖軟體是macOS系統自帶的截圖功能以及macOS版的Snipaste，不知道可不可以在今後的版本中加入監測到特定的快捷鍵時自動隱藏光標的功能，例如我現在使用的command（ALT）+1、command（ALT）+2、command（ALT）+3

**Reply by Blinue** [reply]
目前只会检测 Windows 中的 PrintScreen 键，其他截图方法都是不支持的。切换光标显示的快捷键在 #273 中追踪，目前没有实现计划。

**Comment by Stan650476** [comment]
很棒的程式，但還是想問一下，
請問未來有機會支持全屏顯示以外的放大方式嗎？
例如可以調整大小的視窗化，因為很多老遊戲沒辦法調整視窗大小。

**Reply by hooke007** [reply]
dup https://github.com/Blinue/Magpie/issues/135

**Reply by Stan650476** [reply]
OK, thanks.

**Comment by AHandheldFan** [comment]
Fuck this gay Chinese language shit.
Gonna make my own project similar to this and only going to use English language, you know...just the most universal language in the world.

**Reply by Blinue** [reply]
1. This is a preview version, English support is in plan.
2. Please read [Community Guidelines](https://docs.github.com/en/site-policy/github-terms/github-community-guidelines) before commenting.

**Comment by Ashbringer233** [comment]
缩放时禁用窗口大小调整 每次勾选后关闭Magpie再重新打开又会变回未勾选状态请问可能是什么原因

**Reply by Blinue** [reply]
这是个 bug，已修复。谢谢

**Comment by Violet72965** [comment]
其实你们可以做NVIDA lmage Scaling.简称NIS，也是一种英伟达的超级采样技术，且开源，其实新版做的挺完美了，可以尝试加入新的算法

**Reply by Blinue** [reply]
支持 NIS 和 NVSharpen，在效果菜单里添加即可。旧版本应编辑 ScaleModels.json 添加。
[自定义缩放配置](https://github.com/Blinue/Magpie/wiki/%E8%87%AA%E5%AE%9A%E4%B9%89%E7%BC%A9%E6%94%BE%E9%85%8D%E7%BD%AE)

**Reply by Violet72965** [reply]
我去，我用了好几个版本都没发现，
